src/main/java/org/jenkinsci/plugins/workflow/libs/AbstractGlobalLibraries.java

```diff
@@ -0,0 +1,101 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.workflow.libs;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.model.ItemGroup;
+import hudson.model.Job;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import jenkins.model.GlobalConfiguration;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.kohsuke.stapler.StaplerRequest;
+
+/**
+ * Common code between {@link GlobalLibraries} and {@link GlobalUntrustedLibraries}.
+ */
+public abstract class AbstractGlobalLibraries extends GlobalConfiguration {
+    private List<LibraryConfiguration> libraries = new ArrayList<>();
+
+    protected AbstractGlobalLibraries() {
+        load();
+    }
+
+    public abstract String getDescription();
+
+    public List<LibraryConfiguration> getLibraries() {
+        return libraries;
+    }
+
+    public void setLibraries(List<LibraryConfiguration> libraries) {
+        this.libraries = libraries;
+        save();
+    }
+
+    @Override public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
+        if (Jenkins.get().hasPermission(getRequiredGlobalConfigPagePermission())) {
+            setLibraries(Collections.emptyList()); // allow last library to be deleted
+            return super.configure(req, json);
+        } else {
+            return true;
+        }
+    }
+
+    abstract static class AbstractForJob extends LibraryResolver {
+        protected abstract AbstractGlobalLibraries getConfiguration();
+
+        @NonNull @Override public final Collection<LibraryConfiguration> forJob(@NonNull Job<?,?> job, @NonNull Map<String,String> libraryVersions) {
+            return getLibraries();
+        }
+
+        @NonNull @Override public final Collection<LibraryConfiguration> fromConfiguration(@NonNull StaplerRequest request) {
+            AbstractGlobalLibraries abstractGlobalLibraries = getConfiguration();
+            if (Jenkins.get().hasPermission(abstractGlobalLibraries.getRequiredGlobalConfigPagePermission())) {
+                return getLibraries();
+            }
+            return Collections.emptySet();
+        }
+
+        @NonNull @Override public final Collection<LibraryConfiguration> suggestedConfigurations(@NonNull ItemGroup<?> group) {
+            return getLibraries();
+        }
+
+        private List<LibraryConfiguration> getLibraries() {
+            return getConfiguration()
+                    .getLibraries()
+                    .stream()
+                    .map(this::mayWrapLibrary)
+                    .collect(Collectors.toList());
+        }
+
+        protected LibraryConfiguration mayWrapLibrary(LibraryConfiguration library) {
+            return library;
+        }
```

Comment:
Looking more carefully, it is probably unnecessary to wrap these, see https://github.com/jenkinsci/pipeline-groovy-lib-plugin/pull/129#discussion_r1639938234.